### PR TITLE
feat(createEdit): loaders

### PR DESCRIPTION
**Pull request by flesher1**
Add loaders to CreateEdit sections.

### CODE REVIEW

**Comment by flesher1** [comment]
A lot of small changes makes a big PR. Everything should be working correctly including loaders.

Except Security Marking is broken again. It still needs a state change before you can click it. Not sure how to fix this issue.

File: src/actions/listing.js [hunk 1]
```diff
@@ -51,6 +53,12 @@ export function saveListing(listingData) {
     for(var k in listingData) listingObject[k]=listingData[k];
     var listing = serialize(listingObject);
 
+    if(listing.id === ""){
```
Comment (mannyrivera2010) [inline src/actions/listing.js]: can `listing.id` be `undefined`, `null`? 

datatype is string or integer?

File: src/actions/listing.js [hunk 2]
```diff
@@ -112,7 +148,8 @@ export function fetchActivity(id) {
     };
 }
 
-export function saveImages(images, completedImages) {
+export function saveImages(images, completedImages, listingData) {
+    console.log("saving image")
```
Comment (mannyrivera2010) [inline src/actions/listing.js]: remove console?

File: src/components/createEdit/CreateEdit.js [hunk 3]
```diff
@@ -23,145 +21,76 @@ import Navigation from 'components/sideNavigation/navigation';
 class CreateEdit extends React.Component {
     constructor(props) {
         super(props);
-        this.updateState = this.updateState.bind(this);
-        this.imageUpload = this.imageUpload.bind(this);
-        this.buttonClick = this.buttonClick.bind(this);
         this.state = {
+            id: "",
```
Comment (mannyrivera2010) [inline src/components/createEdit/CreateEdit.js]: Datatype is string or integer

File: src/components/createEdit/CreateEdit.js [hunk 4]
```diff
@@ -23,145 +21,76 @@ import Navigation from 'components/sideNavigation/navigation';
...
-        }
-    }
 
-    updateState(state, value) {
-        var temp = this.state.listing;
-        temp[state] = value;
-        this.setState({listing: temp});
-        this.setState({saved: false})
-    }
+        listingData = Object.assign(listingData, {id: this.state.id});
```
Comment (mannyrivera2010) [inline src/components/createEdit/CreateEdit.js]: Think `id` is integer datatype

File: src/components/createEdit/sections/BasicInformation.js [hunk 5]
```diff
@@ -11,124 +13,177 @@ class BasicInformation extends React.Component {
         this.handleCategoryChange = this.handleCategoryChange.bind(this);
         this.handleTagChange = this.handleTagChange.bind(this);
         this.handleSecMarkChange = this.handleSecMarkChange.bind(this);
+        this.renderForm = this.renderForm.bind(this);
         this.state = {
-            categoriesField: [],
-            tagsField: []
+
```
Comment (mannyrivera2010) [inline src/components/createEdit/sections/BasicInformation.js]: `id` is missing?

**Comment by flesher1** [comment]
Going to keep listing Id as an empty string for now.

**Pull request merged**